classes/hooks/synchronization/Segment.php

```diff
@@ -286,12 +287,13 @@ private function _update($contacts, $existingListId)
                     )
                 );
                 
-                $file = new SplFileObject('contacts.csv', 'w');
+                $file = fopen('contacts.csv', 'w');
```

Comment:
Instead of creating a temporary file you should directly store contacts informations in a variable.